app/controllers/mixins/ems_common_angular.rb

```diff
@@ -429,12 +431,14 @@ def set_ems_record_vars(ems, mode = nil)
         default_endpoint = {:role => :default, :hostname => hostname, :port => port}
         default_endpoint.merge!(endpoint_security_options(ems.security_protocol, default_tls_ca_certs))
 
-        if hawkular_hostname.blank?
-          default_key = params[:default_password] || ems.authentication_key
-          hawkular_hostname = get_hostname_from_routes(ems, default_endpoint, default_key)
+        if params[:event_stream_selection] == 'hawkular_enabled'
```

Comment:
Right, added tests and updated the commit message to be more explicit.
Hawkular enabled => 2 endpoints
Hawkular disabled => 1 endpoint